app/permissions_migrations.go

```diff
@@ -632,5 +728,12 @@ func (a *App) DoPermissionsMigrations() error {
 			return err
 		}
 	}
+	// Need to increase the size of the Permissions column in the Roles table
+	for _, db := range a.Srv().sqlStore.GetAllConns() {
```

Comment:
I spoke to Claudio from our Server team and here is what he had to say:
> You could either increase it or just convert it to `text` and be done with it. In MySQL it would already be `text` anyway since it's more than 255 bytes. I don't see problems either way because I assume the table is usually small. As far as changes, you'd have to change `store/sqlstore/role_store.go` for table creation and add a migration step in `store/sqlstore/upgrade.go`.

In `role_store.go`, you will want to remove the `SetMaxSize` on `Permission`.

In `upgrade.go`, you'll want to write the migration there. You can remove it from here. Also, you can write it only for postgres since MYSQL is already a text. There should already be examples of ALTER in that file for you to follow.

You can see in our gorp library where the decision for text, varchar are happening.
Postgres:
https://github.com/mattermost/gorp/blob/master/dialect_postgres.go#L65

Mysql:
https://github.com/mattermost/gorp/blob/master/dialect_mysql.go#L86